src/provider.rs

```diff
@@ -92,16 +95,24 @@ impl<P: Package, VS: VersionSet> DependencyProvider for OfflineDependencyProvide
             .and_then(|versions| versions.keys().rev().find(|v| range.contains(v)).cloned()))
     }
 
-    type Priority = Reverse<usize>;
+    type Priority = (u32, Reverse<u32>);
```

Comment:
Merging to a `u64` should improve performance.

Reply:
I would've expected LLVM to be able to optimize these to the same thing. Even if not,  I would not expect that  the actual `Ord`  calls are a significant portion of the runtime.  Are you able to actually measure an impact?

Reply:
There was an measurable impact on solana before the improved priorisation because the `PriorityQueue` contained more elements so the Ord impls were called more, but now when using better priorisation the impact is negligible.

Reply:
Comment can be resolved.